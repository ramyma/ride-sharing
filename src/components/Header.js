import React from 'react';
import { useColorMode } from '@xstyled/emotion';
import Flex from './primitives/Flex';
import logo from '../assets/logo.png';
import Box from './primitives/Box';

export default function Header() {
  const [colorMode, setColorMode] = useColorMode();

  function toggleColorMode() {
    if (colorMode === 'dark') {
      setColorMode('default');
    } else {
      setColorMode('dark');
    }
  }

  return (
    <Flex
      as="header"
      height={48}
      grid
      alignItems="center"
      justifyContent="space-between"
      py={2}
    >
      <img height="60%" src={logo} alt="logo" />
      <Box
        backgroundColor="inverseBackground"
        borderRadius="50%"
        clickable
        height={40}
        width={40}
        onClick={toggleColorMode}
      />
    </Flex>
  );
}
