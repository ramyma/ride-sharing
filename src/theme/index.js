import { opacify } from 'polished';

const theme = {
  breakpoints: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200
  },
  defaultColorModeName: 'light',
  colors: {
    background: '#fff',
    grey: '#585858',
    inverseBackground: '#1c1c1c',
    shadow: opacify(0.19, '#000'),
    textDark: '#353535',
    text: '#353535',
    primary: '#ee424a',
    success: '#4CAF50',
    danger: '#f44336',
    warning: '#ffa000',
    modes: {
      dark: {
        background: '#1c1c1c',
        grey: '#9e9e9e',
        inverseBackground: '#fff',
        shadow: opacify(0.19, '#fff'),
        text: '#fff'
      }
    }
  },
  fontSizes: [0, 12, 14, 16, 20, 24, 32, 48, 64, 72],
  sizes: [0, 2, 4, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32],
  shadows: {
    grey: 'rgba(0, 0, 0, 0.19) 0 0 10px 0px',
    greyInset: `inset rgba(0, 0, 0, 0.19) 0 0 10px 0px`,
    pink: '#e42974 0 0 3px',
    text: '0 0 rgba(0, 0, 0, 0.19)'
  },
  space: [0, 4, 8, 16, 24, 48, 96, 144, 192, 240]
};

export default theme;
