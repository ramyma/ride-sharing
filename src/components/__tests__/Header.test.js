import React from 'react';
import renderer from 'react-test-renderer';
import { createSerializer } from 'jest-emotion';
import * as emotion from '@xstyled/emotion';
import { ColorModeProvider, ThemeProvider } from '@xstyled/emotion';
import Header from '../Header';
import theme from '../../theme';

expect.addSnapshotSerializer(createSerializer(emotion));

it('Header renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ColorModeProvider>
          <Header />
        </ColorModeProvider>
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
