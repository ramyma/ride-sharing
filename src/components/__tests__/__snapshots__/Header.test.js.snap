// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.emotion-1 {
  background-color: var(--xstyled-colors-background,#fff);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 48px;
  margin: 0 auto;
  padding: 0 24px;
  padding-top: 8px;
  padding-bottom: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

@media (max-width:1199.98px) {
  .emotion-1 {
    max-width: 100%;
  }
}

@media (min-width:1200px) {
  .emotion-1 {
    max-width: 1200px;
  }
}

.emotion-0 {
  background-color: var(--xstyled-colors-inverseBackground,#1c1c1c);
  border-radius: 50%;
  height: 40px;
  width: 40px;
  cursor: pointer;
}

.emotion-0:disabled {
  cursor: not-allowed;
}

<header
  className="emotion-1"
  height={48}
>
  <img
    alt="logo"
    height="60%"
    src="logo.png"
  />
  <div
    className="emotion-0"
    height={40}
    onClick={[Function]}
    width={40}
  />
</header>
`;
